packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -1,254 +1,241 @@
-        const {
-            addToCart,
-            isMissingOptions,
-            mediaGalleryEntries,
-            productOptions,
-            props
-        } = this;
-        const { classes, isAddingItem, product } = props;
-        const { regularPrice } = product.price;
-
-        // We want this key to change whenever mediaGalleryEntries changes.
-        // Make it dependent on a unique value in each entry (file),
-        // and the order.
-        const carouselKey = mediaGalleryEntries.reduce((fullKey, entry) => {
-            return `${fullKey},${entry.file}`;
-        }, '');
-
-        return (
-            <Form className={classes.root}>
-                <section className={classes.title}>
-                    <h1 className={classes.productName}>
-                        <span>{product.name}</span>
-                    </h1>
-                    <p className={classes.productPrice}>
-                        <Price
-                            currencyCode={regularPrice.amount.currency}
-                            value={regularPrice.amount.value}
-                        />
-                    </p>
-                </section>
-                <section className={classes.imageCarousel}>
-                    <Carousel images={mediaGalleryEntries} key={carouselKey} />
-                </section>
-                <section className={classes.options}>{productOptions}</section>
-                <section className={classes.quantity}>
-                    <h2 className={classes.quantityTitle}>
-                        <span>Quantity</span>
-                    </h2>
-                    <Quantity
-                        initialValue={this.state.quantity}
-                        onValueChange={this.setQuantity}
-                    />
-                </section>
-                <section className={classes.cartActions}>
-                    <Button
-                        priority="high"
-                        onClick={addToCart}
-                        disabled={isAddingItem || isMissingOptions}
-                    >
-                        <span>Add to Cart</span>
-                    </Button>
-                </section>
-                <section className={classes.description}>
-                    <h2 className={classes.descriptionTitle}>
-                        <span>Product Description</span>
-                    </h2>
-                    <RichText content={product.description} />
-                </section>
-                <section className={classes.details}>
-                    <h2 className={classes.detailsTitle}>
-                        <span>SKU</span>
-                    </h2>
-                    <strong>{product.sku}</strong>
-                </section>
-            </Form>
-        );
-    }
-}
+    const handleSelectionChange = useCallback(
+        (optionId, selection) => {
+            // We must create a new Map here so that React knows that the value
+            // of optionSelections has changed.
```

Comment:
I don't follow -- is this some limitation of updating a Map within state? Why would you need to create a new map?

Reply:
When I updated the existing `Map`, the component never re-rendered with the updated `optionSelections`.

I'm assuming that React is doing reference equality here and since the `Map` is the same map (just with different values in it), it thinks nothing changed and therefore doesn't update.

The same as how using `const` with objects works. 

```js
const a = {};
a.b = c;
```

The reference to `a` doesn't change, even though I've changed what `a` contains.

Reply:
I get it. From the docs:

> this is useful when passing callbacks to optimized child components that rely on reference equality to prevent unnecessary renders